Caesars Cipher starter_May1104.py

```diff
@@ -0,0 +1,263 @@
+        
+
+    def get_shift(self):
+        '''
+        Used to safely access self.shift outside of the class
+        
+        Returns: self.shift
+        '''
+        return self.shift
+
+    def get_encrypting_dict(self):
+        '''
+        Used to safely access a copy self.encrypting_dict outside of the class
+        
+        Returns: a COPY of self.encrypting_dict
+        '''
+        return self.encrypting_dict.copy()
+
+    def get_message_text_encrypted(self):
+        '''
+        Used to safely access self.message_text_encrypted outside of the class
+        
+        Returns: self.message_text_encrypted
+        '''
+        return self.message_text_encrypted
+
+    def change_shift(self, shift):
+        '''
+        Changes self.shift of the PlaintextMessage and updates other 
+        attributes determined by shift (ie. self.encrypting_dict and 
+        message_text_encrypted).
+        
+        shift (integer): the new shift that should be associated with this message.
+        0 <= shift < 26
+
+        Returns: nothing
+        '''
+        self.shift = shift
+        self.encrypting_dict = self.build_shift_dict(self.shift)
+        message_text_encrypted = self.apply_shift(self.shift)
+
+class CiphertextMessage(Message):
+    def __init__(self, text):
+        '''
+        Initializes a CiphertextMessage object
+                
+        text (string): the message's text
+
+        a CiphertextMessage object has two attributes:
+            self.message_text (string, determined by input text)
+            self.valid_words (list, determined using helper function load_words)
+        '''
+        Message.__init__(self, text)
+        self.valid_words = self.get_valid_words()
+
+    def decrypt_message(self):
+        '''
+        Decrypt self.message_text by trying every possible shift value
+        and find the "best" one. We will define "best" as the shift that
+        creates the maximum number of real words when we use apply_shift(shift)
+        on the message text. If s is the original shift value used to encrypt
+        the message, then we would expect 26 - s to be the best shift value 
+        for decrypting it.
+
+        Note: if multiple shifts are  equally good such that they all create 
+        the maximum number of you may choose any of those shifts (and their
+        corresponding decrypted messages) to return
+
+        Returns: a tuple of the best shift value used to decrypt the message
+        and the decrypted message text using that shift value
+        '''
+        for number in range(26):
+            decrypted_word = self.apply_shift(26-number)
+            decrypted_word_list = decrypted_word.split()
+            counter = 0
+            for word in decrypted_word_list:
+                if is_word(self.valid_words, word):
+                    counter += 1
+            if counter == len(decrypted_word_list):
+                return (number, decrypted_word)

```

Comment:
This will not work. There might be words in the secret message that are not really english e.g. place names like Bangkok.
Instead you need to count the number of valid words and remember the highest number. @may-nachayada